### PR TITLE
feat(api): populate s51document relational table

### PR DESCRIPTION
**Pull request by Maby0**
## Type of change 🧩

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Other (please explain in the description section above)

## Checklist before requesting a review

- [ ] I have performed a self-review of my own code
- [ ] I have double checked this work does not include any hardcoded secrets or passwords
- [ ] I have made corresponding changes to the documentation
- [ ] I have provided details on how I have tested my code
- [ ] I have referenced the ticket number above
- [ ] I have provided a description of the ticket
- [ ] I have included unit tests to cover any testable code changes

### CODE REVIEW

File: apps/api/src/server/migration/migrators/s51-advice-migrator.js [hunk 1]
```diff
@@ -35,10 +37,20 @@ export const migrateS51Advice = async (s51AdviceList) => {
...
 
-		// TODO attachments
-		// for (const attachmentId in s51AdviceEntity.attachmentIds) {
-		// 	// create attachment/document
-		// }
+		await Promise.all(
+			s51Advice.attachmentIds.map(async (attachmentId) => {
+				return documentRepository.getById(attachmentId).then((documentRow) => {
+					if (!documentRow || !documentRow.guid) {
+						return Promise.reject(new Error(`Failed to get document with ID ${attachmentId}`));
```
Comment (Yakitrak) [inline apps/api/src/server/migration/migrators/s51-advice-migrator.js:44]: Do we need to explicitly reject when we are already throwing error

File: apps/api/src/server/repositories/s51-advice-document.repository.js [hunk 2]
```diff
@@ -74,3 +74,25 @@ export const getPublishedDocumentsByAdviceIds = (s51AdviceIds) => {
...
+			where: { documentGuid: documentId },
+			update: { adviceId: s51AdviceId },
+			create: {
+				documentGuid: documentId,
+				adviceId: s51AdviceId
+			}
+		})
+	];
+
+	const [upsertResult] = await databaseConnector.$transaction(transactionItems);
```
Comment (Yakitrak) [inline apps/api/src/server/repositories/s51-advice-document.repository.js:96]: I don't think we need transaction as its just one prisma query

File: apps/api/src/server/migration/migrators/s51-advice-migrator.js (hunk 1 above)
Comment (Maby0) [reply apps/api/src/server/migration/migrators/s51-advice-migrator.js]: good spot. Fixed

**Review by Yakitrak — DISMISSED** [review]

File: apps/api/src/server/repositories/s51-advice-document.repository.js (hunk 2 above)
Comment (Maby0) [reply apps/api/src/server/repositories/s51-advice-document.repository.js]: Fixed!

**Review by Yakitrak — APPROVED** [review]

**Pull request merged**